src/azure-cli/azure/cli/command_modules/appconfig/_utils.py

```diff
@@ -118,3 +125,38 @@ def prep_null_label_for_url_encoding(label=None):
         label = '"{0}"'.format(label)
         label = ast.literal_eval(label)
     return label
+
+
+def get_appconfig_data_client(cmd, name, connection_string, auth_mode, endpoint):
+    azconfig_client = None
+    if auth_mode == "key":
+        connection_string = resolve_connection_string(cmd, name, connection_string)
+        try:
+            azconfig_client = AzureAppConfigurationClient.from_connection_string(connection_string=connection_string,
+                                                                                 user_agent=HttpHeaders.USER_AGENT)
+        except Exception as ex:
+            raise CLIError("An exception occurred while connecting to App Configuration: {}".format(str(ex)))
```

Comment:
Will AzureAppConfigurationClient.from_connection_string connect with App Configuration? I assume it will just create a client and won't notice failure until it actually sends out requests.

Reply:
AzureAppConfigurationClient.from_connection_string will raise an exception if the connection string is invalid. There is an additional check for empty segments in connection string that we don't check on CLI side. 
Apart from that, you are right, there won't be any exception until a request is sent out. The error message here is more general because I felt that saying something like "failed to create client" is an implementation detail.

Reply:
If the connection string check in SDK failed, then throw it out directly. No need to wrap in a try-catch. The current error message indicates there is something wrong connecting to App Configuration like network or server error. Instead, it is client side's problem. 

---
In reply to: [496238201](https://github.com/Azure/azure-cli/pull/15160#discussion_r496238201) [](ancestors = 496238201)

Reply:
To catch the connection string error, we would still need to wrap this call in try-catch block as we are not using the SDK to validate conn string directly.
How about the error message: "Operation failed to due to an exception: {}"

Reply:
Whatever exception the SDK throws, it is unexpected and can bubble up to clients. For some known errors, we may react on it and wrap it to CLIError instead of catching all exceptions. 

---
In reply to: [496284698](https://github.com/Azure/azure-cli/pull/15160#discussion_r496284698) [](ancestors = 496284698)

Reply:
Fixed